src/include/ipc/compilator_info.h

```diff
@@ -0,0 +1,33 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Karol Trzcinski <karolx.trzcinski@linux.intel.com>
+ */
+
+/**
+ * \file include/ipc/compilator_info.h
+ * \brief Compilator version and name descriptor
+ * \author Karol Trzcinski <karolx.trzcinski@linux.intel.com>
+ */
+
+#ifndef __COMPILATOR_INFO_H__
+#define __COMPILATOR_INFO_H__
+
+#if defined(__XCC__)
+#define CC_MAJOR (XTHAL_RELEASE_MAJOR/1000)
+#define CC_MINOR ((XTHAL_RELEASE_MAJOR%1000)/10)
+#define CC_MICRO XTHAL_RELEASE_MINOR
+#define CC_NAME "XCC"
+/* __XCC__ */
+#elif defined(__GNUC__) && !defined(__XCC__)
```

Comment:
&& !defined(__XCC__) is redundant

Reply:
I do this because when order will be swapt, then it will be needed because XCC base on GNUC. With this extra condition order doesn't matter.